test/test_transforms.py

```diff
@@ -1794,6 +1549,292 @@ def test_randomness(fn, trans, config, p):
+    (torch.Tensor(4, 4, 1).uniform_().numpy(), 'F'),
+    (torch.ByteTensor(4, 4, 1).random_(0, 255).numpy(), 'L'),
+    (torch.ShortTensor(4, 4, 1).random_().numpy(), 'I;16'),
+    (torch.IntTensor(4, 4, 1).random_().numpy(), 'I'),
+])
+def test_1_channel_ndarray_to_pil_image(img_data, mode):
+    for transform in [transforms.ToPILImage(), transforms.ToPILImage(mode=mode)]:
+        img = transform(img_data)
+        assert img.mode == mode
+        # note: we explicitly convert img's dtype because pytorch doesn't support uint16
+        # and otherwise assert_close wouldn't be able to construct a tensor from the uint16 array
+        torch.testing.assert_close(img_data[:, :, 0], np.asarray(img).astype(img_data.dtype))
+
+
+@pytest.mark.parametrize('mode', [None, 'LA'])
+def test_2_channel_ndarray_to_pil_image(mode):
+    img_data = torch.ByteTensor(4, 4, 2).random_(0, 255).numpy()
+
+    if mode is None:
+        img = transforms.ToPILImage()(img_data)
+        assert img.mode == 'LA'  # default should assume LA
+    else:
+        img = transforms.ToPILImage(mode=mode)(img_data)
+        assert img.mode == mode
+    split = img.split()
+    for i in range(2):
+        torch.testing.assert_close(img_data[:, :, i], np.asarray(split[i]), check_stride=False)
+
+
+def test_2_channel_ndarray_to_pil_image_error():
+    img_data = torch.ByteTensor(4, 4, 2).random_(0, 255).numpy()
+    transforms.ToPILImage().__repr__()
+
+    with pytest.raises(ValueError, match=r"Only modes \['LA'\] are supported for 2D inputs"):
+        # should raise if we try a mode for 4 or 1 or 3 channel images
+        transforms.ToPILImage(mode='RGBA')(img_data)
+        transforms.ToPILImage(mode='P')(img_data)
+        transforms.ToPILImage(mode='RGB')(img_data)
+
+
+@pytest.mark.parametrize('mode', [None, 'LA'])
+def test_2_channel_tensor_to_pil_image(mode):
+    img_data = torch.Tensor(2, 4, 4).uniform_()
+    expected_output = img_data.mul(255).int().float().div(255)
+    if mode is None:
+        img = transforms.ToPILImage()(img_data)
+        assert img.mode == 'LA'  # default should assume LA
+    else:
+        img = transforms.ToPILImage(mode=mode)(img_data)
+        assert img.mode == mode
+
+    split = img.split()
+    for i in range(2):
+        assert np.allclose(expected_output[i].numpy(), F.to_tensor(split[i]).numpy())
+
+
+def test_2_channel_tensor_to_pil_image_error():
+    img_data = torch.Tensor(2, 4, 4).uniform_()
+
+    with pytest.raises(ValueError, match=r"Only modes \['LA'\] are supported for 2D inputs"):
+        # should raise if we try a mode for 4 or 1 or 3 channel images
+        transforms.ToPILImage(mode='RGBA')(img_data)
+        transforms.ToPILImage(mode='P')(img_data)
+        transforms.ToPILImage(mode='RGB')(img_data)
+
+
+@pytest.mark.parametrize('data_type, mode', [
+    ('float', 'L'),
+    ('byte', 'L'),
+    ('short', 'I;16'),
+    ('int', 'I'),
+])
+def test_2d_tensor_to_pil_image(data_type, mode):
```

Comment:
Because both `imp_data` and `expected_output` depend on `mode`, I don’t know how to deal with this scenario, so I introduce a new parameter `data_type` to handle this condition statement.

Reply:
I think my comment above answers this but LMK if this isn't clear :)

Reply:
Done as suggested!